KarhooUISDK/Screens/CheckoutScreen/Checkout/KarhooCheckoutPresenter.swift

```diff
@@ -332,18 +337,16 @@ final class KarhooCheckoutPresenter: CheckoutPresenter {
         }
 
         self.trip = trip
-        analytics.bookingRequested(
-            tripDetails: trip,
-            outboundTripId: nil
-        )
-            
+        reportPaymentSuccess()
         view?.showCheckoutView(false)
     }
     
     private func handleGuestAndTokenBookTripResult(_ result: Result<TripInfo>) {
         if let trip = result.successValue() {
+            reportPaymentSuccess()
             callback(.completed(result: trip))
         } else if let error = result.errorValue() {
+            reportPaymentFailure(error.message)
```

Comment:
What if the error occurred independently of payment reasons? Shouldn't we check that the error code is payment related?